packages/amplication-data-service-generator/src/dynamic-plugin-installation.ts

```diff
@@ -0,0 +1,16 @@
+import { ChildProcess, exec } from "child_process";
+
+function promiseFromChildProcess(child: ChildProcess) {
+  return new Promise(function (resolve, reject) {
+    child.addListener("error", reject);
+    child.addListener("exit", resolve);
+  });
+}
+
+export async function dynamicPluginInstallation(
+  pluginInstallations: string[]
+): Promise<void> {
+  await promiseFromChildProcess(
+    exec(`npm i ${pluginInstallations.join(" ")} --no-save -f`)
+  );
```

Comment:
This approach might bring us easily into a dependency hell https://npm.github.io/how-npm-works-docs/theory-and-design/dependency-hell.html.

Plus is locking our solution to only support NPM leaving most used package managers like yarn and pnpm out to support. 
Possibly, a stronger solution would be to bundle all the dependencies of a plugin within the plugin itself.

Reply:
Plus we might have network issues (that are always a possibility) that will affect the installation of npm packages in runtime and, consequently, the code generation.